versioned/persist/adapter/src/main/java/org/projectnessie/versioned/persist/adapter/KeyListEntry.java

```diff
@@ -15,19 +15,29 @@
  */
 package org.projectnessie.versioned.persist.adapter;
 
+import javax.annotation.Nullable;
 import org.immutables.value.Value;
+import org.projectnessie.versioned.Hash;
 import org.projectnessie.versioned.Key;
 
 /** Composite of key, content-id, content-type. */
 @Value.Immutable(lazyhash = true) // this type is used as a map-key in an expensive test
-public interface KeyWithType {
+public interface KeyListEntry {
```

Comment:
is the rename itself really necessary? also javadoc on the class needs to be updated

Reply:
Yea - make it clear that the type represents an entry of `KeyList` and isn't something "generic".